File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.ts
function removeDuplicates(nums: number[]): number {
    let k = 2;
    for(let i = 2; i < nums.length; i ++){
            // console.log(i,k,nums);
        if(nums[i] !== nums[k - 2]){
            nums[k] = nums[i];
            k++;
        }

    }
    return k;
};

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length < 3) return nums.length;
        int keepPosition = 2;
        for(int i = 2; i < nums.length; i ++) {
            if(nums[i] != nums[keepPosition - 2]) nums[keepPosition ++] = nums[i];
        }
        return keepPosition;
    }
}